### ISSUE TITLE
ERROR/Crash: N must be positive?

### ISSUE THREAD

**Issue by Zilacon** [Poweruser/MinetickMod#18]
Its not everyday i see a vanilla server crash, but it did:

```
java.lang.IllegalArgumentException: n must be positive
    at java.util.Random.nextInt(Random.java:300)
    at net.minecraft.server.v1_7_R4.BiomeDecorator.a(BiomeDecorator.java:273)
    at net.minecraft.server.v1_7_R4.BiomeDecorator.a(BiomeDecorator.java:136)
    at net.minecraft.server.v1_7_R4.BiomeBase.a(SourceFile:364)
    at net.minecraft.server.v1_7_R4.BiomeTaiga.a(SourceFile:84)
    at net.minecraft.server.v1_7_R4.ChunkProviderGenerate.getChunkAt(ChunkProviderGenerate.java:332)
    at org.bukkit.craftbukkit.v1_7_R4.generator.NormalChunkGenerator.getChunkAt(NormalChunkGenerator.java:49)
    at net.minecraft.server.v1_7_R4.ChunkProviderServer.getChunkAt(ChunkProviderServer.java:325)
    at net.minecraft.server.v1_7_R4.Chunk.loadNearby(Chunk.java:899)
    at org.bukkit.craftbukkit.v1_7_R4.chunkio.ChunkIOProvider.callStage2(ChunkIOProvider.java:66)
    at org.bukkit.craftbukkit.v1_7_R4.chunkio.ChunkIOProvider.callStage2(ChunkIOProvider.java:13)
    at org.bukkit.craftbukkit.v1_7_R4.util.AsynchronousExecutor$Task.finish(AsynchronousExecutor.java:188)
    at org.bukkit.craftbukkit.v1_7_R4.util.AsynchronousExecutor.finishActive(AsynchronousExecutor.java:347)
    at org.bukkit.craftbukkit.v1_7_R4.chunkio.ChunkIOExecutor.tick(ChunkIOExecutor.java:42)
    at net.minecraft.server.v1_7_R4.MinecraftServer.v(MinecraftServer.java:669)
    at net.minecraft.server.v1_7_R4.DedicatedServer.v(DedicatedServer.java:261)
    at net.minecraft.server.v1_7_R4.MinecraftServer.u(MinecraftServer.java:608)
    at net.minecraft.server.v1_7_R4.MinecraftServer.run(MinecraftServer.java:510)
    at net.minecraft.server.v1_7_R4.ThreadServerApplication.run(SourceFile:628)
```

**Comment by Poweruser** [comment]
The immediate reason for the crash is trivial:
Random.nextInt(int) is passed 0, which is not positive ( >= 1).

The biome decorator can indeed pass 0, when the accessed neighbor chunk is not loaded at that time, because that's when [getHighestBlockYAt(x, z) returns 0](https://github.com/Poweruser/MinetickMod/blob/v1.7.10/src/main/java/net/minecraft/server/World.java#L745).
In order for this not to happen, the method Chunk.loadNearby is supposed to make sure that the neighbor chunks are loaded. In [Chunk.loadNearby](https://github.com/Poweruser/MinetickMod/blob/v1.7.10/src/main/java/net/minecraft/server/Chunk.java#L885) I noticed that in two of the if-clauses identical conditions are tested twice, which is redundant.
In [line 890](https://github.com/Poweruser/MinetickMod/blob/v1.7.10/src/main/java/net/minecraft/server/Chunk.java#L890) the neighbor chunk [i - 1, j + 1] is double checked and
in [line 894](https://github.com/Poweruser/MinetickMod/blob/v1.7.10/src/main/java/net/minecraft/server/Chunk.java#L894) the neighbor chunk [i + 1, j - 1] is double checked.
This raises the question, if there's a mistake in there.
I have to investigate further...

**Comment by Zilacon** [comment]
@Poweruser You just spoke so much alien to me im so confused XD
But it sounds like your on the right track so GJ :+1: 

:)

**Comment by Poweruser** [comment]
Did this happen during at a world load, or during normal gameplay?

**Comment by Zilacon** [comment]
There is no gameplay, its a hub. Pretty much all world generation is disabled, all mobs, pvp, end, nether, villages, everything is disabled. And the hub is in a blank world with no generation or blocks.

And this happened while i was asleep so...

**Comment by Poweruser** [comment]
This bug is quite old and from what i read it happens mostly on clear world servers (Skyblock and such). And it also may happen when all neighbor chunks are loaded, but their heightmap is at 0.

I assume you used some plugin like CleanroomGenerator or EmptyWorldGenerator to create your world, do you still have it your \plugins folder? Because that's the only way I can reproduce it:
- Create an completely empty world with multiverse and one of the said world generators ( /mv create ... -g < generator > )
- Create some chunks in it, then restart the server without the world generator plugin
- Instant crash when a new chunk is generated next to the empty ones

**Comment by Zilacon** [comment]
its still on the server, and im using emptyworldgenerator.

**Comment by Poweruser** [comment]
In 1.8 parts of the chunk decoration are skipped wherever a height map value of 0 is found.
I think I'm going to do that as well instead of using the height 1

**Comment by Zilacon** [comment]
Happened again, big one this time.
http://pastebin.com/Yn9Bb1fQ

&

```
[14:05:47 ERROR]: Exception stopping the server
java.lang.IllegalStateException: Unknown org.bukkit.craftbukkit.v1_7_R4.chunkio.QueuedChunk@d0c289a9
    at org.bukkit.craftbukkit.v1_7_R4.util.AsynchronousExecutor.drop(AsynchronousExecutor.java:266) ~[MineTick.jar:git-Bukkit-1.7.9-R0.2-143-g1dbae1b]
    at org.bukkit.craftbukkit.v1_7_R4.chunkio.ChunkIOExecutor.dropQueuedChunkLoad(ChunkIOExecutor.java:32) ~[MineTick.jar:git-Bukkit-1.7.9-R0.2-143-g1dbae1b]
    at net.minecraft.server.v1_7_R4.PlayerChunk.b(PlayerChunk.java:93) ~[MineTick.jar:git-Bukkit-1.7.9-R0.2-143-g1dbae1b]
    at net.minecraft.server.v1_7_R4.PlayerChunkMap.removePlayer(PlayerChunkMap.java:269) ~[MineTick.jar:git-Bukkit-1.7.9-R0.2-143-g1dbae1b]
    at net.minecraft.server.v1_7_R4.PlayerList.disconnect(PlayerList.java:331) ~[MineTick.jar:git-Bukkit-1.7.9-R0.2-143-g1dbae1b]
    at net.minecraft.server.v1_7_R4.PlayerConnection.a(PlayerConnection.java:706) ~[MineTick.jar:git-Bukkit-1.7.9-R0.2-143-g1dbae1b]
    at net.minecraft.server.v1_7_R4.PlayerConnection.disconnect(PlayerConnection.java:175) ~[MineTick.jar:git-Bukkit-1.7.9-R0.2-143-g1dbae1b]
    at net.minecraft.server.v1_7_R4.PlayerList.u(PlayerList.java:1195) ~[MineTick.jar:git-Bukkit-1.7.9-R0.2-143-g1dbae1b]
    at net.minecraft.server.v1_7_R4.MinecraftServer.stop(MinecraftServer.java:427) ~[MineTick.jar:git-Bukkit-1.7.9-R0.2-143-g1dbae1b]
    at net.minecraft.server.v1_7_R4.MinecraftServer.run(MinecraftServer.java:545) [MineTick.jar:git-Bukkit-1.7.9-R0.2-143-g1dbae1b]
    at net.minecraft.server.v1_7_R4.ThreadServerApplication.run(SourceFile:628) [MineTick.jar:git-Bukkit-1.7.9-R0.2-143-g1dbae1b]
[14:05:47 INFO]: Stopping server
[14:05:47 INFO]: Saving players
```

**Comment by Zilacon** [comment]
Just saw your update, using the new jar now.

**Comment by Poweruser** [comment]
The second exception is a possible result of the failed chunk load.

I still have the suspicion that the custom chunk generator for this world is not working anymore. If you happen to see normal overworld chunks in your clear world now, you know it for sure.

Another way to specify one is in the bukkit.yml (don't mind the missing yaml indentions):
`worlds:`
`Spawn:`
`generator: EmptyWorldGenerator`

Was that maybe, how you had the chunk generator set up? While messing with the bungee/cauldron setup you mentioned once that you did regenerate all server configs. Did this entry (If that was the case) go missing there?

**Comment by Zilacon** [comment]
No i use multiverse so its setup to use that generator through that. And so far it has been completely empty no world generation at all.

**Comment by Poweruser** [comment]
This had been fixed in d2daa42. 0 remains a valid value within the height maps and decorations at height 0 are skipped.
